### ISSUE TITLE
[ BUG ] Build Fail even after setting compileSdkVersion to 30

### ISSUE THREAD

**Issue by sanjaybaskaran01** [Baseflow/flutter-geocoding#74]
## 🐛 Build Fail

```
Note: /home/frosty/.pub-cache/hosted/pub.dartlang.org/geocoding-2.0.1/android/src/main/java/com/baseflow/geocoding/GeocodingPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
/home/frosty/.pub-cache/hosted/pub.dartlang.org/geolocator_android-2.1.0/android/src/main/java/com/baseflow/geolocator/location/LocationMapper.java:29: error: cannot find symbol
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
                                                    ^
  symbol:   variable S
  location: class VERSION_CODES
/home/frosty/.pub-cache/hosted/pub.dartlang.org/geolocator_android-2.1.0/android/src/main/java/com/baseflow/geolocator/location/LocationMapper.java:30: error: cannot find symbol
      position.put("is_mocked", location.isMock());
                                        ^
  symbol:   method isMock()
  location: variable location of type Location
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':geolocator_android:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

### Configuration

**Version:** 
 Flutter 2.5.2 • channel stable
 geolocator: ^7.7.0
 geocoding: ^2.0.1

**Platform:** 
- [ ] :iphone: iOS (unsure)
- [x] :robot: Android 11

### Possible Fix
The following change in build.gradle builds the application

```
compileSdkVersion:31
targetSdkVersion:31
```
Although the following code seems to build the application `placemarkFromCoordinates` method does not work for my system, even for known coordinates it returns 
```
PlatformException(NOT_FOUND, No address information found for supplied coordinates (latitude: 52.216516, longitude: 6.945971)., null, null)
```

**Comment by camelcc** [comment]
Documents needs to be updated. VERSION_CODES.S need sdk 31. Try set `compileSdkVersion 31` and compile again.

**Comment by sanjaybaskaran01** [comment]
> Documents needs to be updated. VERSION_CODES.S need sdk 31. Try set `compileSdkVersion 31` and compile again.

This along with targetSdkVersion:31 compiles but placemarkFromCoordinates method does not work

**Comment by mvanbeusekom** [comment]
Fixed in version 2.0.3
